backend/src/routes/api/profileRoutes.js

```diff
@@ -0,0 +1,23 @@
+import { StatusCodes } from 'http-status-codes';
+import Profile from '../../models/Profile.js';
+
+const userRoutes = async (router) => {
+  router.get('/users/:id', getUser);
+};
+
+const getUser = async (req, res, next) => {
+  try {
+    const userId = req.params.id;
+    const currentProfile = await Profile.findById(userId);
+    if (currentProfile) {
+      res.status(StatusCodes.OK).send(currentProfile);
+    } else {
+      res.status(StatusCodes.BAD_REQUEST).send('User not found.');
+    }
+  } catch (error) {
+    res.status(StatusCodes.BAD_REQUEST).send('Something went wrong.');
+    throw error;
+  }
```

Comment:
I'm wondering if this error will be caught by anything...

If we throw an error in catch block it will be caught by an outer catch block  eg.
```js
try {
    try {
        //something
    } catch(x) {
        throw x;
    }
} catch(x) {
    //handle it
}
```

Reply:
I think it would be more efficient if we just throw an error in try block 
eg. 
instead
```js
res.status(StatusCodes.BAD_REQUEST).send('User not found.');
```

we may 
```js
throw new ExpressError(...) // this is Error object used by our error handling middleware 
```

and then in catch
```
catch (err) {
next(err) // it will invoke error handling middleware
}